parquet-column/src/main/java/org/apache/parquet/column/statistics/Statistics.java

```diff
@@ -175,10 +200,47 @@ public void mergeStatistics(Statistics stats) {
    */
   abstract public void setMinMaxFromBytes(byte[] minBytes, byte[] maxBytes);
 
+  /**
+   * Returns the generic object representing the min value in the statistics. The self comparing logic of the returned
+   * object might not be the proper one (e.g. unsigned comparison for int/long) therefore it is strongly recommended to
+   * use the related comparing method {@link #compareToMin(Object)} or the comparator returned by {@link
+   * #comparator()}.
+   */
   abstract public T genericGetMin();
+
+  /**
+   * Returns the generic object representing the max value in the statistics. The self comparing logic of the returned
+   * object might not be the proper one (e.g. unsigned comparison for int/long) therefore it is strongly recommended to
+   * use the related comparing method {@link #compareToMax(Object)} or the comparator returned by {@link
+   * #comparator()}.
+   */
   abstract public T genericGetMax();
 
   /**
+   * Returns the comparator to be used to compare two generic values in the proper way (e.g. unsigned comparison for
```

Comment:
This implies that unsigned comparison is correct for int. I think you should instead say "For example, unsigned comparison for UINT32"